Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -1358,3 +1400,143 @@ static void WriteColor(ShaderCode& out, APIType api_type, const pixel_shader_uid
     }
   }
 }
+
+static void WriteBlend(ShaderCode& out, const pixel_shader_uid_data* uid_data)
+{
+  if (uid_data->blend_enable)
+  {
+    out.Write("\tfloat4 blend_src;\n");
+    switch (uid_data->blend_src_factor)
```

Comment:
For a smaller code, I'd prefer
`
const char* src_rgb_factors[] = {
  "float3(0,0,0)", // ZERO
  "float3(1,1,1)", // ONE
...
};
out.Write("\tfloat4 blend_src = float4(%s, %s);\n", src_rgb_factors[uid_data->blend_src_factor], src_a_factors[uid_data->blend_src_factor_alpha]);
`
Again, tiny nitpick, feel free to ignore this as well ;)